### ISSUE TITLE
Allow counter increments over TSDs

### ISSUE THREAD

**Issue by tsuna** [OpenTSDB/opentsdb#55]
People frequently need to count discrete observations over time.  Let's say you wanna track ad-clicks.  The recommend way of doing this is to have your application server maintain these counters and write the value of the counter every N seconds to TSDB (preferably through `tcollector`).  In certain cases, this is impractical.  For instance people writing stuff in PHP are having trouble maintaining state across requests to achieve something like this.  They end up maintaining the counter in another system (such as memcache) and having an out-of-band mechanism to collect the counters from that other system and hand them to a TSD every N seconds.  This feature request is about allowing people to do without that separate system and allow them to send counter increments directly to a TSD (any TSD) and let the TSD figure it out and efficiently store the counters in HBase.

What's needed here is a new RPC type for counter increments.  The RPC need to specify the usual stuff (metric, tags) but in addition to the timestamp the TSD need to know, somehow, what interval between data points is wanted.  This way the TSD will maintain the counter (in HBase, using atomic increments) and store its value as a datapoint every N seconds.  We need to define how we want this to work exactly.

**Comment by tcurdt** [comment]
Wow, this is an old issue - and still open?

Not sure I can follow https://github.com/OpenTSDB/opentsdb/pull/184 or https://github.com/OpenTSDB/opentsdb/commit/f53aad648714347431ca07a631b34ce481961f1d

@manolama Is `ZIM` the non-interpolation aggregation one can use for counts as described by @tsuna?

**Comment by manolama** [comment]
@tcurdt Yes, the ZimSum supplies zero if missing so it avoids interpolation. Same with mimmax and mimmin.
